src/karthik/group-people-data/groupPeopleData.ts

```diff
@@ -0,0 +1,41 @@
+interface People {
+  _id: string;
+  name: string;
+  gender: string;
+  email: string;
+  phone: string;
+  city: string;
+}
+
+interface GroupOfPeopleByCity {
+  count: number;
+  city: string;
+  people: Array<People>;
+}
+
+const groupPeopleByCity = (
+  arrayOfObject: Array<People>
+): Array<GroupOfPeopleByCity> => {
+  if (!arrayOfObject || arrayOfObject.length < 1) return [];
+  const grouped = {};
+  const property = 'city';
+  arrayOfObject.forEach((obj) => {
```

Comment:
You have used iteration two time (forEach and map), can we have only one for improved time complexity.

Reply:
clarified.